docs/user-guide/configure.md

```diff
@@ -375,6 +375,35 @@ Patterns are applied against the file path relative to the directory of the conf
 
 Overrides have higher precedence than regular configurations. Multiple overrides within the same config are applied in order. That is, the last override block in a config file always has the highest precedence.
 
+## `processors`
+
+> ⚠️ Warning:
+> This is an experimental feature. The API may change in the future.
+
+Processors are functions that hook into Stylelint's pipeline.
+Currently, processors contains only two properties: a string `name` and a function `postprocess`. `postprocess` runs after all rules have been evaluated. This function receives the `result` object of the linting process and can modify it.
+
+```js
+// post-processor.js
+export default {
```

Comment:
[suggest] I prefer a function module over an object module as follows because:

- A function can change the behavior by the `options` argument (more flexibility)
- A function is similar to the previous processor API, so it may reduce the learning cost
- A function can support async execution, e.g. for dynamic `import()`

```js
export default function myProcessor(options = {}) {
  return {
    name: "...",
    postprocess(result) { /* ... */ }
  };
};
```

Any thoughts?

Reply:
I'm up for changing the object module to function. I'm going to update both documentation and test fixture. 

For 
>A function can change the behavior by the options argument 

Just want to call out that currently I don't have the implementation for accepting options in processors.

Reply:
Here's the [commit](https://github.com/stylelint/stylelint/pull/7568/commits/0ef94c895810bdecea2a86f4828816a835013899) that supports function module processor.

Reply:
I'm thinking a different API is necessary if we want to support adding processor options. 

### Current API:
  The config `processors` type is defined as: `type ConfigProcessors = string | stylelint.Processor | (string | stylelint.Processor)[];`
  And `stylelint.Processor` is the `{name: ..., postprocess: ...}` object.

### API with processor options
To support options to function module processor, the config can look something like:
  ```json
  {
    "processors": [
      { "module": "./path-to-processor.js", "options": { "enabled": true } }
    ]
  }
  ```
  An example of ./path-to-processor.js can be
  ```js
  export default function myProcessor(options = {}) {
    return {
      name: 'remap-location',
      /** @type {(result: LintResult) => void} */
      postprocess(result) { 
        if (options.enabled) { 
          // do something... 
        } 
      }
    };
  }

  ```
  Accordingly, the `processors` API type would be `type ConfigProcessors = string | stylelint.ProcessorConfig | (string | stylelint.ProcessorConfig)[];`
  And `stylelint.ProcessorConfig` is the `{module: ..., options: ...}` object.

### Proposal for initial implementation (this PR)
To keep things simple, how about:
1. initially, we do not support processor options.
2. have just string/string array for config `processors`: `type ConfigProcessors = string | string[];` (or just `string[]` as suggested by another comment). 

This allow us to potentially extend the API later to include processor options, should there for a demand for it.

Reply:
Completely agree 👍🏼 Let's avoid supporting `options` for now and simplify the type as `type ConfigProcessors = string[]`.